great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -287,9 +287,16 @@ def _create_temporary_table(  # noqa: C901 - 18
         else:
             stmt = f'CREATE TEMPORARY TABLE "{temp_table_name}" AS {query}'
         if dialect == GXSqlDialect.ORACLE:
-            try:
-                self._engine.execute(sa.text(stmt_1))
-            except DatabaseError:
-                self._engine.execute(sa.text(stmt_2))
+            with self._engine.connect() as connection:
+                with connection.begin():
+                    try:
+                        connection.execute(sa.text(stmt_1))
+                        stmt = stmt_1
+                    except DatabaseError:
+                        connection.execute(sa.text(stmt_2))
+                        stmt = stmt_2
         else:
-            self._engine.execute(sa.text(stmt))
+            with self._engine.connect() as connection:
+                with connection.begin():
+                    connection.execute(sa.text(stmt))
+        return str(stmt)
```

Comment:
@Shinnnyshinshin I wonder why we now need to return the query, rather than just execute it as before.  Also, it may be problematic in the sense that the statement should probably be compiled, rather than just turn into a string with `str()`.  Thanks.

Reply:
@alexsherstinsky this one was a bit tricky. Because the `connection` object is now scoped within the `with` statement, we don't have access to the `stmt` once we break out of the method. This clashed with how we were testing whether the `schema` was being added correctly. 

If there is a better way to do this, I'd love to discuss. This change was kindof bugging me too, to be honest

Reply:
Update: so the `statement.compile()` step, (which is very helpful) would only be relevant for the query portion of stmt here, and it is already done prior to calling the `_create_temporary_table()` function. So I think the correct update to this method would be to declare a `str` up top (to ensure the `stmt` is declared as a string) and change this portion to `return stmt`. 

The discussion earlier really helped clarify this for me. Thank you @alexsherstinsky 


https://github.com/great-expectations/great_expectations/blob/cf3053deba218a06f8c6a6041274decfd79463f2/great_expectations/execution_engine/sqlalchemy_batch_data.py#L156